### PR TITLE
Create a speaker and a presentation

### PR DESCRIPTION
**Pull request by jw1540**

### CODE REVIEW

File: Resources/Views/Authentication/create_presentation.leaf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#extend("base"):
+    #export("title", "SwiftLeeds")
+        #export("content"):
```
Comment (pdamonkey) [inline Resources/Views/Authentication/create_presentation.leaf:3]: This should be on the same indent as the above export

File: Resources/Views/Authentication/create_speaker.leaf [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#extend("base"):
+    #export("title", "SwiftLeeds")
+        #export("content"):
```
Comment (pdamonkey) [inline Resources/Views/Authentication/create_speaker.leaf:3]: As above

File: Resources/Views/Admin/presentations.leaf [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<h1>Speakers</h1>
```
Comment (pdamonkey) [inline Resources/Views/Admin/presentations.leaf:1]: I was using _ to denote snippets in the home page example, not sure if you want to do the same here.  i.e. _presentations.leaf.  Also, should it be <h1>Presentations</h1>?

File: Sources/App/Model/Event.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+//  
+//
+//  Created by Joe Williams on 10/04/2022.
+//
+
+import Foundation
+import Fluent
+import Vapor
+
+final class Event: Model, Content, Codable {
```
Comment (pdamonkey) [inline Sources/App/Model/Event.swift:12]: Do you still need Codable if you have Content?

File: Sources/App/Model/Event.swift [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    
+    typealias IDValue = UUID
+    
+    static var schema: String {
```
Comment (pdamonkey) [inline Sources/App/Model/Event.swift:16]: Could this just be `static let schema = "events"`?

File: Sources/App/Model/Event.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    
+    @Field(key: "location")
+    var location: String
+    
+    @Children(for: \.$event)
+    var presentations: [Presentation]
+    
+    init() { }
+    
+    class Migrations: AsyncMigration {
```
Comment (pdamonkey) [inline Sources/App/Model/Event.swift:37]: Might be worth putting migrations into their own files in a Migrations directory?

File: Sources/App/Auth/Controllers/SpeakerController.swift [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    }
+    
+    private func onPostCreate(request: Request) async throws -> Response {
+        let speaker = try request.content.decode(Speaker.self)
+        let image = try request.content.decode(ImageUpload.self)
+        
+        let app = request.application
+        let filename = "\(image.profileImage.filename)-\(UUID.generateRandom().uuidString)"
+        let path = app.directory.publicDirectory + "img/\(filename)"
+        let handle = try app.fileio.openFile(path: path,
```
Comment (SwiftyAlex) [inline Sources/App/Auth/Controllers/SpeakerController.swift:31]: This is very nitpicky but I normally make this a smidge more readable with:

```
try app.fileio.openFile(
      path: path,
      mode: .write,
      flags: .allowFileCreation(posixMode: 0x744),
      eventLoop: request.eventLoop
).wait()
```

File: Sources/App/Auth/Controllers/SpeakerController.swift [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                                             mode: .write,
+                                             flags: .allowFileCreation(posixMode: 0x744),
+                                             eventLoop: request.eventLoop).wait()
+        try app.fileio.write(fileHandle: handle, buffer: image.profileImage.data, eventLoop: request.eventLoop).wait()
```
Comment (SwiftyAlex) [inline Sources/App/Auth/Controllers/SpeakerController.swift:35]: Can we use the async versions of these APIs ? I think theyre around now ( no big if not ) [docs](https://docs.vapor.codes/4.0/files/)

File: Sources/App/Model/Speaker.swift [hunk 9]
```diff
@@ -2,15 +2,60 @@
 //  Speaker.swift
 //  
 //
-//  Created by Alex Logan on 19/03/2022.
+//  Created by Joe Williams on 09/04/2022.
```
Comment (SwiftyAlex) [inline Sources/App/Model/Speaker.swift:5]: 😢

**Review by SwiftyAlex — APPROVED** [review]
Looks good to me, just some tiny suggestions but nothing I'd say is needed, very nice work 🚀

File: Sources/App/Model/Speaker.swift (hunk 9 above)
Comment (jw1540) [reply Sources/App/Model/Speaker.swift:5]: haha I don't know how this happened, maybe I recreated it! sorry!!

**Pull request merged**